gumbo-parser/src/error.c

```diff
@@ -347,13 +347,14 @@ static void handle_parser_error (
       if (error->parser_state == GUMBO_INSERTION_MODE_INITIAL) {
         print_message(output, "You must provide a doctype");
       } else {
-        print_message(output, "Premature end of file");
+        print_message(output, "Premature end of file.");
         print_tag_stack(error, output);
       }
       return;
     case GUMBO_TOKEN_START_TAG:
     case GUMBO_TOKEN_END_TAG:
-      print_message(output, "That tag isn't allowed here");
+      print_message(output, "'%s' tag isn't allowed here.",
+                    gumbo_normalized_tagname(error->input_tag));
```

Comment:
Would it be better to have the error message make clear if it's a start or end tag? Maybe `<%s> isn't allowed here.` or `</%s> isn't allowed here.`? Or maybe that'd be confusing if the opening tag is really something like `<foo attr1="value1" ... attrn="valuen">` and the error message says `<foo>`

Maybe `Start tag '%s' isn't allowed here.` and `End tag '%s' isn't allowed here.` would be better?

Reply:
Oooh, I like that. :+1:

Reply:
Updated the PR.